scripts/src/tests/helpers.ts

```diff
@@ -182,24 +181,40 @@ export async function createOffers() {
 	}
 }
 
-async function createMalformedOffer(num: number) {
-	const offerName = generateId();
+export async function createMalformedOffer(): Promise<string> {
+	const offerName = "malformed";
 	await createEarn(
-		`${ offerName }_earn${ num }`,
+		offerName,
 		"GBOQY4LENMPZGBROR7PE5U3UXMK22OTUBCUISVEQ6XOQ2UDPLELIEC4J",
-		`earn${ num }`, `earn${ num }`, `earn${ num }`, `earn${ num }`, 100, 30, 1, `earn${ num }`, `earn${ num }`, "poll", animalPoll, ["ALL"]
+		"earn", "earn", "earn", "earn",
+		100,
+		30,
+		1,
+		"earn", "earn",
+		"poll", animalPoll,
+		["ALL"]
 	);
 
-	const offerContents = (await OfferContent.findOne({ where: { name: offerName } }))!;
-	await offerContents.save(); // TypeORM parses the offers content json and saves it back to the table (it gets in not escaped) - such offer content is malformed and has to be filtered out in every .getAll
+	const offer = (await Offer.findOne({ where: { name: offerName } }))!;
+	const malformedOfferContent = (await OfferContent.findOne({ where: { offerId: offer.id } }))!;
 
-	/*	sql for check whether the db has one malformed and wellformed offer content
-		```
-			select * from (select * from offer_contents where content like '{%' limit 1)
-			union all
-			select * from (select * from offer_contents order by id limit 1)
-		```
-	 */
+	malformedOfferContent.content = "{a: 1}";
+	await malformedOfferContent.save();
+
+	return offer.id;
+}
+
+export async function createAppUser(offer: Offer, appId: string): Promise<User> {
```

Comment:
It's a weird name for what this function does once you removed it from the context. Maybe rename this createAppUserWithOffer

Reply:
I just moved existed inner function (closure) from another place
I'll change its name